reference-implementation/to-upstream-wpts/transform-streams/brand-checks.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+  self.importScripts('../resources/test-utils.js');
+}
+
+const TransformStreamDefaultController = getTransformStreamDefaultControllerConstructor();
+
+function getTransformStreamDefaultControllerConstructor() {
+  return realTSDefaultController().constructor;
+}
+
+function fakeTS() {
+  return Object.setPrototypeOf({
+    get readable() { return new ReadableStream(); },
+    get writable() { return new WritableStream(); }
+  }, TransformStream.prototype);
+}
+
+function realTS() {
+  return new TransformStream();
+}
+
+function fakeTSDefaultController() {
+  return Object.setPrototypeOf({
+    get desiredSize() { return 1; },
+    enqueue() { },
+    close() { },
+    error() { }
+  }, TransformStreamDefaultController.prototype);
+}
+
+function realTSDefaultController() {
+  let controller;
+  new TransformStream({
+    start(c) {
+      controller = c;
+    }
+  });
+  return controller;
+}
+
+test(() => {
+  getterThrowsForAll(TransformStream.prototype, 'readable',
+                     [fakeTS(), realTSDefaultController(), undefined, null]);
+}, 'TransformStream.prototype.readable enforces a brand check');
+
+test(() => {
+  getterThrowsForAll(TransformStream.prototype, 'writable',
+                     [fakeTS(), realTSDefaultController(), undefined, null]);
+}, 'TransformStream.prototype.writable enforces a brand check');
+
+test(() => {
+  constructorThrowsForAll(TransformStreamDefaultController,
+                          [fakeTS(), realTS(), realTSDefaultController(), undefined, null]);
+}, 'TransformStreamDefaultConstructor enforces a brand check and doesn\'t permit independant construction');
```

Comment:
Nit: "independ*e*nt". (I have trouble with this a lot so I imagine you probably copied this from a typo I made elsewhere.)